src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+                            __apply_truncation_process()
+                                __cut_list_and_get_list_capacity()
+                                    __apply_truncation()
+                                    
+                            __recompose_truncated_status_file()                                    
+                                __get_current_complete_status_errors()
+                                __recompose_truncated_msg()
+                                    __recompose_substatus_errors()
+                                    __create_assessment_tombstone_list()
+                                        __create_assessment_tombstone()
+                                    __recreate_assessment_summary_json()
+                           
+                            __recompose_truncated_status_file()                                    
+                                __get_current_complete_status_errors()
+                                __recompose_truncated_msg()
+                                    __recompose_substatus_errors()
+                                    __create_installation_tombstone
+                                    __recreate_installation_summary_json()
+                                    
+                            current_package_list_size -= diff
+                            if current_package_list_size <= max_package_list_capacity:
+                                break
+                            __get_new_size_in_bytes_after_truncation() 
+                    
+                write_with_retry_using_temp_file()  
+        """
+        status_file_size_in_bytes = len(complete_status_file_payload.encode("utf-8"))   # calc complete_status_file_payload byte size on disk
+        truncated_status_file = json.loads(complete_status_file_payload)    # reload payload into python object
+
+        if status_file_size_in_bytes > self.__internal_file_capacity:   # 126kb
+            self.composite_logger.log_debug("Begin package list truncation")
+            low_pri_index = None
+            packages_capacity_diff = 0
+            assessment_substatus_index = self.__get_substatus_index(Constants.PATCH_ASSESSMENT_SUMMARY, truncated_status_file['status']['substatus'])
+            installation_substatus_index = self.__get_substatus_index(Constants.PATCH_INSTALLATION_SUMMARY, truncated_status_file['status']['substatus'])
+
+            if assessment_substatus_index is not None:        # if assessment data exists
+                self.__assessment_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, assessment_substatus_index)
+                self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+            if installation_substatus_index is not None:      # if installation data exists
+                low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy, low_pri_index)
```

Comment:
shouldn't this line come after line 940? i.e. when self.__installation_packages_copy is initialized

Reply:
yes, you're right, good catch on this. this will perform the full installation truncation until a follow up operation (like status change) then will use the initialized __installation_packages_copy for high and low pri truncation